internal/database/access_tokens.go

```
@@ -406,6 +447,19 @@ func toSHA256Bytes(input []byte) []byte {
 	return b[:]
 }
 
+func (s *accessTokenStore) logAccessTokenDeleted(ctx context.Context, deletionType SecurityEventName, arg []byte) {
+	a := actor.FromContext(ctx)
+
+	securityEventStore := NewDBWith(s.logger, s).SecurityEventLogs()
+	securityEventStore.LogEvent(ctx, &SecurityEvent{
+		Name:      deletionType,
+		UserID:    uint32(a.UID),
```

I was caught out by `LogEvent` silently failing if `a.UID` was a zero value while writing unit tests. Do you think it's worth adding a warning log line in `LogEvent` when this happens?

---

Yeah, I'd do that. In theory, we should always have the actor ID available, but if there was some kind of background job pruning stale tokens using internal actor the call would crash.

---

Realised what's going on - it _does_ log if the insert fails, but that type of logging isn't printed to screen during unit test runs (at least in the context I was running the tests).

https://github.com/sourcegraph/sourcegraph/blob/will/auditlog/security-events/access-token-deleted/internal/database/security_event_logs.go#L144